sam.c

```diff
@@ -2150,8 +2150,8 @@ int sam_parse1(kstring_t *s, sam_hdr_t *h, bam1_t *b)
     if (*p != '*') {
         uint32_t *cigar = NULL;
         int old_l_data = b->l_data;
-        uint32_t n_cigar = bam_parse_cigar(p, &p, b);
-        if (!n_cigar || *p++ != '\t') goto err_ret;
+        int32_t n_cigar = bam_parse_cigar(p, &p, b);
```

Comment:
Being nit-picky this should be ssize_t, but frankly the world will have broken long before we get to 2 billion cigar elements!  Infact I'm not sure it's possible in BAM as the bam record would overflow in length.

I'd just go with an unsized "int" or a specific "ssize_t" if you feel inclined to be pedantic.